### ISSUE TITLE
[pickers] DatePicker displays empty calendar with minDate in future and timezone UTC

### ISSUE THREAD

**Issue by alex-hrb-0** [mui/mui-x#14453]
### Steps to reproduce

Link to live example: https://codesandbox.io/p/sandbox/zsyzzx

Steps:
1. DatePicker + dayjs. No default value, no initail value.
2. Set `timezone="UTC"` and `minDate` to future (at least tomorrow)
3. Open calendar

### Current behavior

The calendar is empty, literally:
![image](https://github.com/user-attachments/assets/6d4feade-eb09-410f-a80b-ccebdfefba01)
The future months as well

### Expected behavior

![image](https://github.com/user-attachments/assets/25886ed9-3fd2-4a17-98c4-eeaa4c2c2bf9)

### Context

I'm trying to use DatePicker with `timezone="UTC"` and `minDate`. Sometimes I have `minDate` with value in the future. 

**Search keywords**: DatePicker, minDate, timezone

**Search keywords**:

**Comment by noraleonte** [comment]
Hey @alex-hrb-0 👋 
Thanks for raising this 🙏 I double checked and it seems you are right. When the timezone is set and minDate is a future date, the calendar does not display correctly. This only seems to be the case when there is no value set 🤔 I am adding this to the board to discuss & prioritize with the team 🛩️

**Comment by michelengelen** [comment]
Hey @alex-hrb-0 ... this is a duplicate of #13875 and #9653.
The dayjs package seems to be broken (https://github.com/iamkun/dayjs/issues/2303)
Thanks for opening anyways.

**Comment by alex-hrb-0** [comment]
@noraleonte, @michelengelen,
Thank you guys for the reply

> Hey @alex-hrb-0 ... this is a duplicate of #13875 and #9653. The dayjs package seems to be broken ([iamkun/dayjs#2303](https://github.com/iamkun/dayjs/issues/2303)) Thanks for opening anyways.

Yeah, I thought the same until I tried other DateTime libs. It works only with `date-fns` which is basically functions for native JS Date type. I updated the [sandbox example](https://codesandbox.io/p/sandbox/zsyzzx) as well.

**Comment by michelengelen** [comment]
Thanks for the update @alex-hrb-0 ... seems like we marked this as a duplicate preemptively.
I'll re-add this to the board. 👍🏼

**Comment by alex-hrb-0** [comment]
@michelengelen,
I made a mistake in my test example: I left `timezone="UTC"` in a generic component wrapper so it used `dayjs` approach for setting UTC for DatePicker for all tested time libraries. I retested it again: it works perfectly fine with `luxon`. I didn't figure out how to set up UTC with `moment` w\o initial value. With `date-fns` it's impossible to set up UTC  at all. 

So `luxon` is the way out for me. And it feels that you were right about problems in `dayjs` package.
